lib/routes/me.js

```diff
@@ -89,20 +89,29 @@ module.exports = function(crowi, app) {
         return res.render('me/index', {});
       }
 
-      userData.update(name, email, lang, function(err, userData) {
-        if (err) {
-          for (var e in err.errors) {
-            if (err.errors.hasOwnProperty(e)) {
-              req.form.errors.push(err.errors[e].message);
-            }
-          }
+      User.findOne({email: email}, (err, existingUserData) => {
+        if (existingUserData !== null) {
+          debug('Email address was duplicated');
+          req.form.errors.push('It can not be changed to that mail address');
           return res.render('me/index', {});
         }
 
-        req.i18n.changeLanguage(lang);
-        req.flash('successMessage', req.t('Updated'));
-        return res.redirect('/me');
+        userData.update(name, email, lang, (err, userData) => {
+          if (err) {
+            for (var e in err.errors) {
```

Comment:
You should write `Object.keys(err.errors).forEach(() => ...)` and delete L:102.

Reply:
Thanks! I'll fix.